source/grid/manifold.cc

```diff
@@ -25,8 +25,26 @@ DEAL_II_NAMESPACE_OPEN
 
 using namespace Manifolds;
 
-/* -------------------------- Manifold --------------------- */
+// This structure is used as comparison function to std::sort when sorting
+// points accordings to their weight.
+struct CompareWeights
+{
+public:
+  CompareWeights(const std::vector<double> &weights)
+  {
+    compare_weights = &weights;
```

Comment:
I would have just done this initialization in the member initializer list. But this is ok as well.